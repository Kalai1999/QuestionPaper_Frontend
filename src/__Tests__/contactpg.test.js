import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import App from '../Views/contact';
import {create} from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { Button } from 'reactstrap';

Enzyme.configure({ adapter: new Adapter() });

describe("Form Testing",()=>{
    test ("Matches the snapshot",()=>{
        const form = create(<App />);
        expect(form.toJSON()).toMatchSnapshot();        
    });

    test("calls sent method when send button is clicked",()=>{
        const submitHandler=jest.fn();
        const wrapper = mount (<Button onClick={submitHandler}/>);
        const form = wrapper.find('Button');
        form.simulate('click');
        expect(submitHandler.mock.calls.length).toEqual(1);
    });

    test('checking inputs',()=>{
        let wrapper;
        window.alert = jest.fn();
        wrapper=shallow(<App />);
        wrapper.find('StyledTextField').at(0).simulate('change',{target:{name:'name',value:'testing'}});
        wrapper.find('StyledTextField[type="email"]').at(1).simulate('change',{target:{name:'email',value:'testing@gmail.com'}});
        wrapper.find('StyledTextField[type="text"]').at(2).simulate('change',{target:{name:'msg',value:'testing'}});
        expect(wrapper.state('name')).toEqual('testing');
        expect(wrapper.state('email')).toEqual('testing@gmail.com');
        expect(wrapper.state('msg')).toEqual('testing');
        wrapper.find('Button').simulate('click');
        expect(window.alert.mock.calls.length).toBe(1);
    });

});