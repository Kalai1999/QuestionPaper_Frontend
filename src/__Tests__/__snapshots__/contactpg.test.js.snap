exports[`Form Testing Matches the snapshot 1`] = `
<div>
  <div
    className="header">
    <h1>
      We will be happy to hear from you !
    </h1>
    <h4>
      Get in touch with us.
    </h4>
  </div>
  <div
    className="container">
    <div
      className="indentCorrection card">
      <div
        className="card-body">
        <h5>
          Use the public key
        </h5>
        <h2
          className="violet">
          D16C 2228 B32C EA4A B6B8 B92B EE4D DD8F DA96 7765
        </h2>
        <h5>
          to mail us at teamtomato.oss@gmail.com
        </h5>
      </div>
    </div>
    <h3
      className="center">
      or
    </h3>
    <form
      className="">
      <input
        className="form-control"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Your Name"
        sm={10}
        type="text"
        value="" />
      <div
        className="invalid-feedback">
        
      </div>
      <br />
      <input
        className="form-control"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="youremail@email.com"
        sm={10}
        type="email"
        value="" />
      <div
        className="invalid-feedback">
        
      </div>
      <br />
      <textarea
        className="form-control"
        name="message"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Your Message"
        sm={10}
        value="" />
      <div
        className="invalid-feedback">
        
      </div>
      <br />
      <button
        aria-label={null}
        className="btn btn-secondary btn-block"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "violet",
          }
        }
        type="button"
        variant="contained">
        Send
      </button>
    </form>
  </div>
</div>
`;
